pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -458,7 +481,7 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		}
 	}
 
-	// At this point, all of the current Replicas are Running and Ready, we can consider termination.
+	// At this point, all of the current Replicas are Running and Ready or Available, we can consider termination.
```

Comment:
Ready and Available, you can't be one or the other, `IsPodAvailable` checks `IsPodReady` in the first place.